File: scripts/process-pool/process-pool.py
#!/usr/bin/env python3

import os
from concurrent.futures import ProcessPoolExecutor

def worker(task_id):
    name = os.getpid()
    print(f"Process {os.getpid()} doing task: {task_id}")

if __name__ == '__main__':
    with ProcessPoolExecutor(max_workers=5) as pool:
        for task_id in range(10):
            pool.submit(worker, task_id)

File: scripts/process-pool/process-pool.pl
#!/usr/bin/env perl

use v5.38;
use MCE;

my $mce = MCE->new(
    max_workers => 5,
    chunk_size  => 1,
    user_func   => sub {
        my ($mce, $chunk_ref, $chunk_id) = @_;
        my $task_id = $chunk_ref->[0];
        say "Process ", $mce->wid, " doing task: $task_id";
    },
);

$mce->process([0..9]);
$mce->shutdown;
